File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
import java.util.*;

class Solution {
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = schedules.length;
        
        for(int s = 0; s < schedules.length; s++) {
            int safeTime = getPlusTime(schedules[s], 10);
            int day = startday;
            
            // 해당 인원의 타임로그
            for(int timelog : timelogs[s]) {
                if(5 >= day && safeTime < timelog) {
                    // 이벤트 실패!
                    answer--;
                    break;
                }
                // 날짜 증가
                day = day % 7 + 1;
            }
        }
        
        return answer;
    }
    
    public int getPlusTime(int time, int add) {
        int minute = time % 100 + add;
        int hour = time / 100 + minute / 60;
        minute %= 60;
        return hour * 100 + minute;
    }
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.kt
class Solution {

    fun solution(schedules: IntArray, timelogs: Array<IntArray>, startday: Int): Int {
        var answer: Int = 0

        var targetTimes = timelogs[0].size
        var nowDay = startday
        val notLateCntList = IntArray(timelogs.size) { 0 }

        for(y in 0 .. timelogs[0].size - 1) {

            if(nowDay < 6) {

                for(x in 0 .. timelogs.size - 1) {

                    notLateCntList[x] = if(schedules[x].isNotLate(timelogs[x][y])) { 
                        notLateCntList[x] + 1 
                    }else { 
                        notLateCntList[x] 
                    }
                }

            }else {
                targetTimes--
            }

            nowDay = if(++nowDay > 7) { 
                nowDay % 7 
            }else { 
                nowDay 
            }

        }

        for(i in 0 .. notLateCntList.size - 1) {

            if(notLateCntList[i] == targetTimes) { answer++ }
        }

        return answer
    }

    fun Int.isNotLate(commuteTime: Int): Boolean {
        val plusedTime = this + 10

        val chkHours = plusedTime / 100 + if((plusedTime % 100) / 60 > 0) { 1 } else { 0 }
        val chkMinutes = (plusedTime % 100) % 60

        return (chkHours > commuteTime / 100) || (chkHours == commuteTime / 100 && chkMinutes >= commuteTime % 100)
    }
}